prometheus_client/context_managers.py

```diff
@@ -50,12 +50,21 @@ def _new_timer(self):
 
     def __enter__(self):
         self._start = default_timer()
+        return self
 
     def __exit__(self, typ, value, traceback):
         # Time can go backwards.
         duration = max(default_timer() - self._start, 0)
+        instance = self._callback.__self__
+        instance._raise_if_not_observable()
         self._callback(duration)
 
+    def labels(self, *args, **kw):
+        instance = self._callback.__self__
```

Comment:
I don't love the references back to `__self__` and `__name__`. I wonder if we could avoid that, but haven't come up with a great way yet. Perhaps a second variable to `__init__` could help?